components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/DynamicX509TrustManager.java

```diff
@@ -0,0 +1,141 @@
+/*
+*  Copyright (c) 2005-2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.identity.core.util;
+
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.base.ServerConfiguration;
+import org.wso2.carbon.base.api.ServerConfigurationService;
+
+
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import java.io.*;
+import java.security.KeyStoreException;
+import java.security.KeyStore;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+
+/**
+ * Gives a singleton javax.net.ssl.X509TrustManager implementation that uses the default carbon trust store.
+ * This will load any changes (addition/removal of certificates) done to the default trust store on the fly.
+ */
+public class DynamicX509TrustManager implements X509TrustManager {
+    private static Log log = LogFactory.getLog(DynamicX509TrustManager.class);
+    private X509TrustManager trustManager;
+    private static DynamicX509TrustManager instance;
+    //System Properties
+    public static final String PROP_TRUST_STORE_UPDATE_REQUIRED = "org.wso2.carbon.identity.core.util.DynamicX509TrustManager.TRUST_STORE_UPDATE_REQUIRED";
+
+    private DynamicX509TrustManager() throws Exception{
+        setupTrustManager();
+    }
+
+    public static DynamicX509TrustManager getInstance() throws Exception {
```

Comment:
Usually we don't throw Exception or CertificateException, for each component we have defined specific types of exceptions, eg-: Identity-core -IdentityException, You should wrap whatever exceptions you are throwing with such exception, so ideally this should throw IdentityException with a meaning full message.